contracts/treasury/priceOracle/KyberDmmChainLinkPriceOracle.sol

```diff
@@ -0,0 +1,382 @@
+      quoteEthProxyDecimals = quoteEthProxies[i] == address(0) ? 0 :
+        IChainLinkAggregatorProxy(quoteEthProxies[i]).decimals();
+      quoteUsdProxyDecimals = quoteUsdProxies[i] == address(0) ? 0 :
+        IChainLinkAggregatorProxy(quoteUsdProxies[i]).decimals();
+
+      _tokenData[tokens[i]] = AggregatorProxyData({
+        quoteEthProxy: quoteEthProxies[i],
+        quoteUsdProxy: quoteUsdProxies[i],
+        quoteEthProxyDecimals: quoteEthProxyDecimals,
+        quoteUsdProxyDecimals: quoteUsdProxyDecimals
+      });
+    }
+  }
+
+  function updatePremiumBps(address[] calldata liquidators, uint256[] calldata premiumBps)
+    external override onlyAdmin
+  {
+    require(liquidators.length == premiumBps.length, 'invalid length');
+    for(uint256 i = 0; i < liquidators.length; i++) {
+      _setGroupPremiumBps(liquidators[i], premiumBps[i]);
+    }
+  }
+
+  function updateWhitelistedTokens(address[] calldata tokens, bool isAdd)
+    external onlyAdmin
+  {
+    _updateWhitelistedToken(tokens, isAdd);
+  }
+
+  /**
+   * @dev Return list of min amounts that expected to get in return
+   *  when liquidating corresponding list of src tokens
+   *  3 types for hint: REMOVE_LIQUIDITY, LIQUIDATE_TOKENS, LIQUIDATE_LP
+   *  - REMOVE_LIQUIDITY: Take a single LP token, and return 2 tokens in the pool
+   *  - LIQUIDATE_TOKENS: Take list of tokens (must not be whitelisted tokens), then
+   *      liquidate them to a single whitelisted token
+   *  - LIQUIDATE_LP: Take list of LP tokens, then liquidate them to a single whitelisted token
+   *  Apply premium discount, can be a different value for each liquidator.
+   * @param liquidator address of the liquidator
+   * @param tokenIns list of src tokens
+   * @param amountIns list of src amounts
+   * @param tokenOuts list of return tokens
+   * @param hint hint for getting conversion rates
+   * @return minAmountOuts min expected amount for each token out
+   */
+  function getExpectedReturns(
+    address liquidator,
+    IERC20Ext[] calldata tokenIns,
+    uint256[] calldata amountIns,
+    IERC20Ext[] calldata tokenOuts,
+    bytes calldata hint
+  )
+    external override view
+    returns (uint256[] memory minAmountOuts)
+  {
+    require(tokenIns.length == amountIns.length, 'invalid length');
+    minAmountOuts = new uint256[](tokenOuts.length);
+
+    (OracleHintType hintType) = abi.decode(hint, (OracleHintType));
+
+    // Remove Liquidity given a LP token
+    if (hintType == OracleHintType.REMOVE_LIQUIDITY) {
+      require(tokenIns.length == 1, 'invalid number token in');
+      require(tokenOuts.length == 2, 'invalid number token out');
+      (IERC20Ext[2] memory tokens, uint256[4] memory amounts) = getExpectedTokensFromLp(
+        address(tokenIns[0]), amountIns[0]
+      );
+      if (tokens[0] == tokenOuts[0]) {
+        require(tokens[1] == tokenOuts[1], 'invalid token out 1');
+        (minAmountOuts[0], minAmountOuts[1]) = (amounts[2], amounts[3]);
+      } else {
```

Comment:
should it be `else if (tokens[0] == tokenOuts[1])` instead? then revert for else case